cmd/put.go

```diff
@@ -60,68 +62,96 @@ func uploadChunked(dbx files.Client, r io.Reader, commitInfo *files.CommitInfo,
 }
 
 func put(cmd *cobra.Command, args []string) (err error) {
-	if len(args) == 0 || len(args) > 2 {
-		return errors.New("`put` requires `src` and/or `dst` arguments")
+	if len(args) == 0 {
+		return errors.New("missing operands to `put`")
 	}
 
-	src := args[0]
-
-	// Default `dst` to the base segment of the source path; use the second argument if provided.
-	dst := "/" + path.Base(src)
-	if len(args) == 2 {
-		dst, err = validatePath(args[1])
-		if err != nil {
-			return
-		}
-	}
-
-	contents, err := os.Open(src)
-	defer contents.Close()
+	destination, err := cmd.Flags().GetString("destination")
 	if err != nil {
-		return
+		return err
 	}
 
-	contentsInfo, err := contents.Stat()
-	if err != nil {
-		return
-	}
-
-	progressbar := &ioprogress.Reader{
-		Reader: contents,
-		DrawFunc: ioprogress.DrawTerminalf(os.Stderr, func(progress, total int64) string {
-			return fmt.Sprintf("Uploading %s/%s",
-				humanize.IBytes(uint64(progress)), humanize.IBytes(uint64(total)))
-		}),
-		Size: contentsInfo.Size(),
-	}
-
-	commitInfo := files.NewCommitInfo(dst)
-	commitInfo.Mode.Tag = "overwrite"
-
-	// The Dropbox API only accepts timestamps in UTC with second precision.
-	commitInfo.ClientModified = time.Now().UTC().Round(time.Second)
-
-	dbx := files.New(config)
-	if contentsInfo.Size() > chunkSize {
-		return uploadChunked(dbx, progressbar, commitInfo, contentsInfo.Size())
+	var waitGroup sync.WaitGroup
+	for _, arg := range args {
+		waitGroup.Add(1)
+		go func(arg string) error {
```

Comment:
Consider defining a helper that takes a waitGroup as arg. Will reduce indent here and make it easier to understand what the high-level flow is.